docs/src/references/mappers/c8y-mapper.md

```diff
@@ -1325,3 +1325,191 @@ te/device/main///cmd/log_upload/<cmd_id>
+    Child device ->> MQTT Broker: Telemetry message to `te/device/plc1///m/temperature`
+    MQTT Broker ->> Tedge Mapper: Telemetry message to `te/device/plc1///m/temperature`
+    Tedge Mapper ->> Tedge Mapper: Auto-register the `plc1` device with auto-generated `external-id`
+    MQTT Broker ->> Tedge Mapper: Deliver child device registration message on `te/device/plc1//` `{ "id": "desired-xid" }`
+    Tedge Mapper ->> Tedge Mapper: Ignore the registration message as the entity is already registered
+```
+
+Here the end-result is even more catastrophic, as the user expects a device with external-id: `desired-xid`,
+but the mapper has created one with the auto-generated external-id, which can't be changed afterwards.
+
+Here is another variant of the same issue, with nested child devices, for whom explicit registration is mandatory:
+
+```mermaid
+sequenceDiagram
+    participant Child device
+    participant MQTT Broker
+    participant Tedge Mapper
+    
+    Child device ->> MQTT Broker: Child device registration message to `te/device/nested1//` `{ "@parent": "device/child1//" }`
+    Child device ->> MQTT Broker: Telemetry message to `te/device/nested1///m/temperature`
+    MQTT Broker ->> Tedge Mapper: Telemetry message to `te/device/nested1///m/temperature`
+    Tedge Mapper ->> Tedge Mapper: Auto-register the `nested1` device as immediate child device
+    MQTT Broker ->> Tedge Mapper: Deliver child registration message on `te/device/nested1//` `{ "@parent": "device/child1//" }`
+    Tedge Mapper ->> Tedge Mapper: Ignore the child registration message as the entity is already registered
+```
+
+Here a device that was meant to be registered as a nested child device got registered as immediate child device,
+which can't be undone/corrected when the actual registration message eventually arrives.
+
+#### Initial solution thoughts
+
+One obvious solution to avoid this problem is to force the users to disable auto-registration,
+when they intend to use explicit registration even for a single entity.
+But the downside with that approach is that we force them to use explicit registration for everything,
+even the entities for which auto-registration would have been perfectly fine,
+just because a few entities required some customizations via explicit registration.
+
+The other approach would be to resort to auto-registration only when
+an explicit registration message is not delivered within a predefined window(timeout).
+But, the same timing uncertainty issues described above, applies here as well:
+what if the explicit registration message is received after the auto-registration has already happened after the timeout?
+Crashing itself isn't gonna help here either, as the entity is already auto-registered.
+
+### Final Solution
+
+The ordering problems when the mapper is live and when it is restarted would be handled differently.
+
+#### Phase 1
+
+**Live mapper**
```

Comment:
This sounds as a simple mitigation which can be improved in the future without breaking users' configuration files or expectations. I have to admit that my preferred solution (a user-defined regex-based topic scheme) is a commitment for a feature that will have to be maintained.